osu.Game.Rulesets.Osu/Edit/Blueprints/Sliders/Components/PathControlPointPiece.cs

```diff
@@ -195,13 +196,34 @@ private void updateMarkerDisplay()
 
             markerRing.Alpha = IsSelected.Value ? 1 : 0;
 
-            Color4 colour = ControlPoint.Type.Value != null ? colours.Red : colours.Yellow;
+            Color4 colour = getColourFromNodeType();
 
             if (IsHovered || IsSelected.Value)
                 colour = colour.Lighten(1);
 
             marker.Colour = colour;
             marker.Scale = new Vector2(slider.Scale);
         }
+
+        private Color4 getColourFromNodeType()
+        {
+            if (!(ControlPoint.Type.Value is PathType pathType))
+                return colours.Yellow;
+
+            switch (pathType)
+            {
+                case PathType.Catmull:
+                    return colours.Seafoam;
+
+                case PathType.Bezier:
+                    return colours.Pink;
+
+                case PathType.PerfectCurve:
+                    return colours.PurpleDark;
```

Comment:
feels a bit too dark than other colours, what about:
```suggestion
                    return colours.Purple;
```
before:
<img width="240" alt="image" src="https://user-images.githubusercontent.com/22781491/111739942-e85edb00-8894-11eb-9e5a-b0a80bce3106.png">

after:
<img width="249" alt="image" src="https://user-images.githubusercontent.com/22781491/111740456-db8eb700-8895-11eb-83e3-4276fa5d6928.png">

Reply:
I was trying to make the lightness levels slightly different because I guarantee there's some colour-blindness mode where these two are indistinguishable (purple and pink seem quite close together).

The lighter one looks better but not sure.

Reply:
Oh didn't take colour-blindness modes into account, yeah I think it's fine for now to keep distinguishing between each.